integration-tests/test_register.py

```diff
@@ -0,0 +1,73 @@
+import pytest
+import functools
+import logging
+
+logger = logging.getLogger(__name__)
+
+pytestmark = pytest.mark.usefixtures("register_subman")
+
+
+def test_register(insights_client):
+    """
+    Test `insights-client --register` when the client is registered
+    """
+    response = insights_client.run("--register", check=False)
+    assert (
+        response.returncode == 0
+    ), "An application returns 0 after successful registration"
+    assert (
+        "Successfully registered host" in response.stdout
+    ), "An application should inform that a system was registered"
+    assert (
+        "Starting to collect Insights data" in response.stdout
+    ), "An application should inform that Insights data will be collected"
+    assert (
+        "Successfully uploaded report" in response.stdout
+    ), "An application should inform that report was successfully uploaded"
+    assert (
+        "View the Red Hat Insights console" in response.stdout
+    ), "An application should inform about Insights console"
+    assert (
+        insights_client.is_registered
+    ), "Current state of registration should be 'registered'"
+
+
+def test_register_twice(insights_client):
+    """
+    Test `insights-client --register` when the client is registered
+    """
+    insights_client.run("--register")
+    assert insights_client.is_registered
+
+    response = insights_client.run("--register")
+    assert (
+        "This host has already been registered" in response.stdout
+    ), "An application should inform that a system was already registered"
+    assert (
+        "Successfully uploaded report" in response.stdout
+    ), "An application should inform that report was successfully uploaded"
+    assert (
+        "View the Red Hat Insights console" in response.stdout
+    ), "An application should inform about Insights console"
+    assert (
+        insights_client.is_registered
+    ), "Current state of registration should be 'registered'"
```

Comment:
this test is now covered by `test_double_registration` (provided by #191); if extra checks are needed, I'd add them to that test

Reply:
the test 'test_double_registration' plays with machine-id. It is the main point of the test - the machine-id is not changed anytime. A test in this PR verifies user interface behaviour rather. I think it makes sense to keep those two tests separated.

Reply:
> the test 'test_double_registration' plays with machine-id.

... because that was its first implementation, and there is nothing that limits it to machine-id checks; that tests is really about double registration, so it makes sense to have all the checks related to double registration attempts there: machine-id checks, output checks, etc

Reply:
merged.